### PR TITLE
common access for multi features coordinates, lines and rings as traits

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
@jdroenner you wanted to have a common access trait for these types.
While unfortunately, we can't have the same type (unless you tell me how to convert a Vec<Vec<X>> to &[&[X]] without creating a new vector on the fly) but I managed to find common `AsRef`s.

### CODE REVIEW

File: datatypes/src/primitives/multi_point.rs [hunk 1]
```diff
@@ -4,6 +4,11 @@ use crate::util::arrow::ArrowTyped;
 use crate::util::Result;
 use snafu::ensure;
 
+/// A trait that allows a common access to lines of `MultiPoint`s and its references
```
Comment (michaelmattig) [inline datatypes/src/primitives/multi_point.rs:7]: copy paste

Comment (ChristianBeilschmidt) [reply datatypes/src/primitives/multi_point.rs:7]:
```suggestion
/// A trait that allows a common access to lines of `MultiPolygons`s and its references
```

**Comment by jdroenner** [comment]
This is a very good idea to unify access to different multi feature implementations. Could you write tests for it? Also, i guess a benchmark would be helpful to see if there is a performance penalty when using the as_ref indirection.

**Comment by ChristianBeilschmidt** [comment]
@jdroenner I've added tests.

I guess I don't have to benchmark the `as_ref` call for now. We should discuss our benchmarking separately, fix a current version, and then see improvements or degradations on further changes.

**Pull request merged**